x/blob/keeper/keeper_test.go

```diff
@@ -27,34 +29,44 @@ import (
 
 // TestPayForBlobs verifies the attributes on the emitted event.
 func TestPayForBlobs(t *testing.T) {
-	k, _, ctx := CreateKeeper(t)
-	signer := "celestia15drmhzw5kwgenvemy30rqqqgq52axf5wwrruf7"
-	namespace := share.MustNewV0Namespace(bytes.Repeat([]byte{1}, share.NamespaceVersionZeroIDSize))
-	namespaces := [][]byte{namespace.Bytes()}
-	blobData := []byte("blob")
-	blobSizes := []uint32{uint32(len(blobData))}
-
-	// verify no events exist yet
-	events := ctx.EventManager().Events().ToABCIEvents()
-	assert.Len(t, events, 0)
-
-	// emit an event by submitting a PayForBlob
-	msg := createMsgPayForBlob(t, signer, namespace, blobData)
-	_, err := k.PayForBlobs(ctx, msg)
-	require.NoError(t, err)
+	// Testing transition from v2 -> v3
+	// v2 uses the GasPerBlobByte key from the params store
+	// v3 uses the GasPerBlobByte from the appconsts
+	// TODO: Replace this with appversion latest
+	versions := []uint64{v2.Version, v3.Version}
```

Comment:
[question] why only test appversion latest instead of all 3 app versions?

```suggestion
	// TODO: Replace this with appversion latest
	versions := []uint64{v1.Version, v2.Version, v3.Version}
```

Reply:
we don't really need v1.Version here because we care that the transitionary versions work. For example for versions v2 and v1 it's the same behaviour and they'd fall under the same if statement same goes for versions v3+. I don't think it'd be sustainable to have all versions hardcoded in tests.

Reply:
On re-read, I find this test confusing b/c it doesn't actually make any assertions for version 2 and version 3. I think this test would be useful if it looked something like:

```go
	type testCase struct {
		version uint64
		want    int64
	}
	testCases := []testCase{
		{version: v1.Version, want: 10},
		{version: v2.Version, want: 8},
	}
```

but since it doesn't do that, we could just make it non-version aware:

```go
// TestPayForBlobs verifies the attributes on the emitted event.
func TestPayForBlobs(t *testing.T) {
	k, _, ctx := CreateKeeper(t, v3.Version)
	signer := "celestia15drmhzw5kwgenvemy30rqqqgq52axf5wwrruf7"
	namespace := share.MustNewV0Namespace(bytes.Repeat([]byte{1}, share.NamespaceVersionZeroIDSize))
	namespaces := [][]byte{namespace.Bytes()}
	blobData := []byte("blob")
	blobSizes := []uint32{uint32(len(blobData))}

	// verify no events exist yet
	events := ctx.EventManager().Events().ToABCIEvents()
	assert.Len(t, events, 0)

	// emit an event by submitting a PayForBlob
	msg := createMsgPayForBlob(t, signer, namespace, blobData)
	_, err := k.PayForBlobs(ctx, msg)
	require.NoError(t, err)

	// verify that an event was emitted
	events = ctx.EventManager().Events().ToABCIEvents()
	assert.Len(t, events, 1)
	protoEvent, err := sdk.ParseTypedEvent(events[0])
	require.NoError(t, err)
	event, err := convertToEventPayForBlobs(protoEvent)
	require.NoError(t, err)

	// verify the attributes of the event
	assert.Equal(t, signer, event.Signer)
	assert.Equal(t, namespaces, event.Namespaces)
	assert.Equal(t, blobSizes, event.BlobSizes)
}
```

which I realize now is probably what you meant by the TODO.

Reply:
yeah exactly. I tried to avoid hardcoding the version. In hindsight I would've only started working on this after the app was properly upgraded to v3. Do you think it'd be better if I update the application to v3 and then I can also address the issues in tests.